waggle-dance-api/src/test/java/com/hotels/bdp/waggledance/api/model/AbstractMetaStoreTest.java

```diff
@@ -104,4 +110,39 @@ public void emptyName() {
     Set<ConstraintViolation<T>> violations = validator.validate(metaStore);
     assertThat(violations.size(), is(1));
   }
+
+  @Test
+  public void equalsNull() {
+    assertFalse(metaStore.equals(null));
+  }
+
+  @Test
+  public void equalsDifferentClass() {
+    assertFalse(metaStore.equals("string"));
+  }
+
+  @Test
+  public void newFederatedInstance() {
+    FederatedMetaStore federatedMetaStore = metaStore.newFederatedInstance(name, remoteMetaStoreUri);
```

Comment:
static methods is better called in a static manner. `metaStore. newFederatedInstance` -> `AbstractMetaStore. newFederatedInstance`